File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] arr, int val) {
        
        int n = arr.length, l = 0, r = n-1;
        
        while(l <= r){
            if(arr[r] == val){
                r--;
                continue;
            }
            if(arr[l] == val){
                int t = arr[r];
                arr[r] = arr[l];
                arr[l] = t;
                r--;
            }
            l++;
        }
        // System.out.println(l + "-" + r);
        return l;
    }
}

File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& a, int val) {
        int c=0,n=a.size();
        int i=-1;
        for(int j=0;j<n;j++){
          if(a[j]!=val){
              c++;
              i++;
              swap(a[i],a[j]);
          }   
        }
        return c;
    }
};